.github/workflows/conformance-ginkgo.yaml

```diff
@@ -0,0 +1,926 @@
+
+          - k8s-version: "1.25"
+            focus: "datapath-egress-tun-vxlan-endpoint-routes-disabled"
+
+          - k8s-version: "1.25"
+            focus: "datapath-egress-tun-vxlan-endpoint-routes-enabled"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ns-tc"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ns-misc"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ns-xdp-1"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ns-xdp-2"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ew-1"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ew-2"
+
+          - k8s-version: "1.25"
+            focus: "datapath-service-ew-kube-proxy"
+
+          - k8s-version: "1.25"
+            focus: "datapath-bgp-custom-lrp"
+
+          - k8s-version: "1.25"
+            focus: "kafka"
+
+          # These tests require an external node which is only available on 1.27
+          # / net-next so there's no point on running them
+          - k8s-version: "1.19"
+            focus: "agent-policy-multi-node-2"
+
+          # These tests are always skipped so there's no point on running
+          - k8s-version: "1.19"
+            focus: "datapath-egress-tun-disabled-endpoint-routes-disabled"
+
+          - k8s-version: "1.19"
+            focus: "datapath-egress-tun-disabled-endpoint-routes-enabled"
+
+          - k8s-version: "1.19"
+            focus: "datapath-egress-tun-vxlan-endpoint-routes-disabled"
+
+          - k8s-version: "1.19"
+            focus: "datapath-egress-tun-vxlan-endpoint-routes-enabled"
+
+          - k8s-version: "1.19"
+            focus: "datapath-service-ns-tc"
+
+          - k8s-version: "1.19"
+            focus: "datapath-service-ns-misc"
+
+          - k8s-version: "1.19"
+            focus: "datapath-service-ns-xdp-1"
+
+          - k8s-version: "1.19"
+            focus: "datapath-service-ns-xdp-2"
+
+    steps:
+      - name: Set up job variables
+        id: vars
+        run: |
+          if [ ${{ github.event.issue.pull_request || github.event.pull_request }} ]; then
+            PR_API_JSON=$(curl \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+              ${{ github.event.issue.pull_request.url || github.event.pull_request.url }})
+            SHA=$(echo "$PR_API_JSON" | jq -r ".head.sha")
+          else
+            SHA=${{ github.sha }}
+          fi
+
+          echo "sha=${SHA}" >> $GITHUB_OUTPUT
+
+          ipfamily="dual"
+          case ${{ matrix.k8s-version }} in
+              1.27)
+                  kube_image="kindest/node:v1.27.1@sha256:b7d12ed662b873bd8510879c1846e87c7e676a79fefc93e17b2a52989d3ff42b"
+                  kernel="bpf-next-20230526.105339@sha256:4133d4e09b1e86ac175df8d899873180281bb4220dc43e2566c47b0241637411"
```

Comment:
Yes. Having the kernel and kube_image as part of the matrix it so would be considered a two new dimensions in the matrix and increase the number of combinations to test.

Reply:
But can this not be handled via `include`, such that the matrix does not have 2 dimensions but the proper kernel env variables are inserted for the proper K8s version? 🤔

Reply:
Ah, also one of my first approaches :D After working with it for a couple days it was really frustrating to maintain the workflow https://github.com/cilium/cilium/actions/runs/4573014881/workflow#L175-L497

Reply:
Discussed offline with @nbusseneau about this. I've tested out @nbusseneau proposal [here](https://github.com/cilium/cilium/actions/runs/5158740999/jobs/9292789247) and it works really well. 

Unfortunately, GitHub workflow files lack a GitHub context that allows access to the job name during their execution. This presents a significant challenge when generating the JUnit filename because it is generated based on the job name and matrix name. On datastudio, we retrieve the job name for the GitHub API, which typically appears as something like `"name": "E2E Test (1.26, datapath-bgp-custom-lrp)"`. However, the JUnit filename incorporates all matrix dimensions and looks like this: `E2E Test (1.26, datapath-bgp-custom-lrp, 5.4-20230526.105339@sha256:523ff0c81ed9be73a2d0d02a64c72655225efdd32e35220f530ca3eff3eada3c).xml`. This discrepancy makes it exceedingly difficult to establish a clear correlation between the JUnit filename and the job name. We need to know which JUnit file was generated by which GitHub job so that we can store it properly on datastudio.

Reply:
This is because we set up the filenames for sysdump and JUnit with `${{ join(matrix.*, '-') }}`:
```
./cilium-cli sysdump --output-filename cilium-sysdump-${{ join(matrix.*, '-') }}-final || true
...
junit_filename="${{ env.job_name }} (${{ join(matrix.*, ', ') }}).xml"
```
To get back to the same behaviour as initially proposed in the PR, we can use `${{ matrix.k8s-version }}-${{ matrix.focus }}`:
```
./cilium-cli sysdump --output-filename cilium-sysdump-${{ matrix.k8s-version }}-${{ matrix.focus }}-final || true
...
junit_filename="${{ env.job_name }} (${{ matrix.k8s-version }}-${{ matrix.focus }}).xml"
```